tests/src/test/scala/cats/tests/FunctionTests.scala

```diff
@@ -42,19 +42,25 @@ class FunctionTests extends CatsSuite {
 
 
   // law checks for the various Function0-related instances
-  checkAll("Function0[Eqed]", OrderLaws[Function0[Eqed]].eqv)
-  checkAll("Function0[POrd]", OrderLaws[Function0[POrd]].partialOrder)
-  checkAll("Function0[Ord]", OrderLaws[Function0[Ord]].order)
-  checkAll("Function0[Hsh]", HashLaws[Function0[Hsh]].hash)
-  checkAll("Function0[Semi]", GroupLaws[Function0[Semi]].semigroup)
-  checkAll("Function0[CSemi]", GroupLaws[Function0[CSemi]].commutativeSemigroup)
-  checkAll("Function0[Bnd]", GroupLaws[Function0[Bnd]].band)
-  checkAll("Function0[SL]", GroupLaws[Function0[SL]].semilattice)
-  checkAll("Function0[BSL]", GroupLaws[Function0[BSL]].boundedSemilattice)
-  checkAll("Function0[Mono]", GroupLaws[Function0[Mono]].monoid)
-  checkAll("Function0[CMono]", GroupLaws[Function0[CMono]].commutativeMonoid)
-  checkAll("Function0[Grp]", GroupLaws[Function0[Grp]].group)
-  checkAll("Function0[CGrp]", GroupLaws[Function0[CGrp]].commutativeGroup)
+  checkAll("Function0[Eqed]", EqLawTests[Function0[Eqed]].eqv)
+  checkAll("Function0[POrd]", PartialOrderLawTests[Function0[POrd]].partialOrder)
+  checkAll("Function0[Ord]", OrderLawTests[Function0[Ord]].order)
+  checkAll("Function0[Semi]", SemigroupLawTests[Function0[Semi]].semigroup)
+  checkAll("Function0[CSemi]", CommutativeSemigroupTests[Function0[CSemi]].commutativeSemigroup)
+  checkAll("Function0[Bnd]", BandTests[Function0[Bnd]].band)
+  checkAll("Function0[SL]", SemilatticeTests[Function0[SL]].semilattice)
+  checkAll("Function0[BSL]", BoundedSemilatticeTests[Function0[BSL]].boundedSemilattice)
+  checkAll("Function0[Mono]", MonoidLawTests[Function0[Mono]].monoid)
+  checkAll("Function0[CMono]", CommutativeMonoidTests[Function0[CMono]].commutativeMonoid)
+  checkAll("Function0[Grp]", GroupLawTests[Function0[Grp]].group)
+  checkAll("Function0[CGrp]", CommutativeGroupTests[Function0[CGrp]].commutativeGroup)
+
+  test("Function0[Hsh]") {
```

Comment:
Just curious, why not just test the full HashLaw? won't pass?

Reply:
Yeah, it's not the same as universal hash and not the same as `scala.util.Hashing`